src/compile/axis.ts

```diff
@@ -222,15 +226,14 @@ export function tickSizeEnd(model: Model, channel: Channel) {
   return undefined;
 }
 
-
 export function title(model: Model, channel: Channel) {
   const axis = model.axis(channel);
   if (axis.title !== undefined) {
     return axis.title;
   }
 
   // if not defined, automatically determine axis title from field def
-  const fieldTitle = fieldDefTitle(model.fieldDef(channel));
+  const fieldTitle = fieldDefTitle(axisFieldDef(model, channel));
```

Comment:
I am not entirely sure why you think so, can you explain here or on slack?
The only problem here was that our axis is always called `x` or `y` and we use them as `channel` to refer to the `x-axis` or `y-axis`. Now since `x-axis` has 2 channels `x` and `x2` which makes our logic of using `X` as the channel to get their `fieldDef` incorrect. When there is no `x2` the `fieldDef(X)` doesn't exist so we need to make sure we are getting the `x2` fieldDef